### PR TITLE
chore(tests): fix some sporiadic test failures

### PR DESCRIPTION
**Pull request by skateman**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: test/factories/benchmark.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
     title { Faker::Lorem.sentence }
     description { Faker::Lorem.paragraph }
     ref_id { "xccdf_org.ssgproject.content_benchmark_RHEL-#{os_major_version}" }
-    sequence(:version) { |n| "0.#{(n / 50).floor}.#{n % 50}" }
+    sequence(:version) { |n| "100.#{(n / 50).floor}.#{n % 50}" }
```
Comment (skateman) [inline test/factories/benchmark.rb:8]: Our benchmarks are in the format of `0.y.z` which sometimes matched a random generated here.

File: test/factories/test_result.rb [hunk 2]
```diff
@@ -5,6 +5,6 @@
     host { association :host }
     profile { association :profile, :with_rules }
     end_time { Time.zone.now }
-    score { SecureRandom.rand(100) }
+    score { SecureRandom.rand(98) + 1 }
```
Comment (skateman) [inline test/factories/test_result.rb:8]: Sometimes the random here generated a 0 or 100 score that was not working well with some of our tests.

File: test/graphql/queries/system_query_test.rb [hunk 3]
```diff
@@ -769,7 +769,7 @@ class SystemQueryTest < ActiveSupport::TestCase
 
     setup_two_hosts
 
-    WHost.limit(1).update(tags: [TAG])
+    WHost.update(tags: [TAG])
```
Comment (skateman) [inline test/graphql/queries/system_query_test.rb:772]: The `limit(1)` is nondeterministic, setting both hosts with the tags is fine.

**Comment by romanblanco** [comment]
Attempt to fix sporadic test failure:

> https://github.com/RedHatInsights/compliance-backend/runs/5434680127?check_suite_focus=true

```
Progress: |========================ERROR["test_: update_os_minor_versions child profiles multiple supported benchmarks with older assigned version should not touch the newer profile. ", #<Minitest::Reporters::Suite:0x00007fa2ec1f0838 @name="PolicyTest">, 28.441981718999983]
 test_: update_os_minor_versions child profiles multiple supported benchmarks with older assigned version should not touch the newer profile. #PolicyTest (28.44s)
Minitest::UnexpectedError:         ActiveRecord::RecordInvalid: Validation failed: Ref has already been taken
            test/models/policy_test.rb:406:in `block (4 levels) in <class:PolicyTest>'
            test/models/policy_test.rb:418:in `instance_exec'
            test/models/policy_test.rb:418:in `block in create_test_from_should_hash'

============================================|
```

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

**Pull request merged**